packages/twenty-front/src/modules/workspace-invitation/hooks/useInviteUser.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  useSendInviteLinkMutation,
+  SendInviteLinkMutationVariables,
+} from '~/generated/graphql';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+import { workspaceInvitationsState } from '../states/workspaceInvitationsStates';
+
+export const useInviteUser = () => {
+  const [sendInviteLinkMutation] = useSendInviteLinkMutation();
+
+  const setWorkspaceInvitations = useSetRecoilState(workspaceInvitationsState);
+  const workspaceInvitations = useRecoilValue(workspaceInvitationsState);
+
+  const sendInvitation = async (emails: SendInviteLinkMutationVariables) => {
+    return await sendInviteLinkMutation({
```

Comment:
here you don't really need to LISTEN workspaceInvitations : you only need it when you set the state as you want to append to the state.

To do that, the setWorkspaceInvitations has a syntax to expose the currentState value
setWorkspaceInvitations((workspaceInvitations) => { my logic })

Even if setWorkspaceInvitations was not exposing workspaceInvitations in a callback we would not listen to invitations and would prefer useRecoilCallback which is exposing the current state snapshot without listening to them

Listening to states is an issue as it will cause re-renders (here to any component using useInviteUser